wurst/objects/abilities/beastMaster/Druid/SpiritualGuidance.wurst

```diff
@@ -89,45 +89,46 @@ public function toggleSpiritForm(unit pet, bool inRange)
 
     // If the unit is out of range and is a spirit.
     else if not inRange and SPIRITS.has(pet)
-        // Return the pet to it's normal color and transparency.
-        pet.setVertexColor(255, 255, 255, 255)
+        removeSpiritForm(pet)
 
-        // Reduce the damage by 2.
-        pet.setBaseDamage(pet.getBaseDamage(0) - PET_STAT_BONUS , 0)
+function removeSpiritForm(unit pet)
+    // Return the pet to it's normal color and transparency.
+    pet.setVertexColor(255, 255, 255, 255)
 
-        // Reduce the armor by 2.
-        pet.setArmor(pet.getArmor() - PET_STAT_BONUS)
+    // Reduce the damage by 2.
+    pet.setBaseDamage(pet.getBaseDamage(0) - PET_STAT_BONUS , 0)
 
-        // Reduce magic resistance by 2 levels.
-        pet.setMagicResist(pet.getMagicResist() - PET_STAT_BONUS)
+    // Reduce the armor by 2.
+    pet.setArmor(pet.getArmor() - PET_STAT_BONUS)
 
-        // Remove this unit from the list of spirits.
-        SPIRITS.remove(pet)
+    // Reduce magic resistance by 2 levels.
+    pet.setMagicResist(pet.getMagicResist() - PET_STAT_BONUS)
+
+    // Remove this unit from the list of spirits.
+    SPIRITS.remove(pet)
 
 // Used to continuously check if the pet is in range of the aura.
 function checkPetRadius()
     doPeriodically(PET_PROXIMITY_INTERVAL) (CallbackPeriodic cb) ->
-        // Cycle through all the owner/pet combos.
-        for owner in pets
+        pets.forEach() (player key, unit value) ->
```

Comment:
I think the variables name are fine here, but just in case you didn't know, there's no need to use the defaults from autocomplete. For example,  `pets.forEach() (player owner, unit pet)` would work and reduce the required changes.